STAGING - ID changes (#1941)
OpenLiberty/blogs: posts/2021-11-30-microprofile-5-22001-beta.adoc

[#mp_5.0]
=== MicroProfile 5.0

MicroProfile 5.0 enables applications to use MicroProfile APIs together with Jakarta EE 9.1. MicroProfile 5.0 does not provide any other functional updates except aligning with Jakarta EE 9.1.  MicroProfile 5.0 includes the following features:

* Config 3.0
* Fault Tolerance 4.0
* Rest Client 3.0 
* Health 4.0
* Metrics 4.0
* Open Tracing 3.0
* Open API 3.0
* JWT propagation 2.0. 

This beta driver will be used as a compatible implementation for releasing MicroProfile 5.0.

> Include the following code in your `server.xml` file...

Reviewer: ```suggestion
Include the following in the `server.xml` to enable the MicroProfile 5.0 feature in Open Liberty:
```

</featureManager>
----

For more information about this update, check out the link:https://github.com/eclipse/microprofile/releases/tag/5.0[MicroProfile 5.0 Release] on GitHub.

[#rest_client]
=== MicroProfile Rest Client 3.0

> MicroProfile Rest Client is an API that helps developers to write type-safe interfaces that abstract and invoke remote RESTful services. This is the 3.0 release of MicroProfile Rest Client, and it adds support for Jakarta EE 9.1 technologies. From a developer's perspective, the only change from the previous release (2.0) is the package space name change of Jakarta packages from `javax.\*` to `jakarta.*`. Another change is that the Open Liberty implementation has changed from Apache CXF to RESTEasy - this change brings with it some behavior and property changes (most of which are already documented as link:{url-prefix}/docs/latest/reference/jaxrs-dif.html[differences between `jaxrs-2.1` and `restfulWS-3.0`]).  

Reviewer: ```suggestion
MicroProfile Rest Client is an API that helps developers to write type-safe interfaces that abstract and invoke remote RESTful services. This is the 3.0 release of MicroProfile Rest Client, and it adds support for Jakarta EE 9.1 technologies. From a developer's perspective, the only change from the previous release (2.0) is the package space name change of Jakarta packages from `javax.*` to `jakarta.*`. However, another change is that the Open Liberty implementation has changed from Apache CXF to RESTEasy - this change brings with it some behavior and property changes (most of which are already documented as link:{url-prefix}/docs/latest/reference/jaxrs-dif.html[differences between `jaxrs-2.1` and `restfulWS-3.0`]).  
```

[#jpa_persistence]
=== Define JPA persistence properties at server scope

This new JPA enhancement adds the ability to declare "default" JPA persistence properties to all container-managed persistence contexts as a Liberty server.xml configuration.

Previously if a persistence property needed to be set for all persistence.xml configuration files you would manually update all persistence.xml files in all applications. This could end up requiring hundreds of manual updates and/or rebuilding of applications.
With this enhancement you can specify persistence properties in the server.xml that will propagate to all container-managed persistence units for applications installed on that server.

> To start using the new feature, add the `<defaultProperties>` configuration element to the `<jpa>` configuration in your `server.xml` file. Specify the persistence properties that you want to apply to all container-managed persistence units, as shown in the following example:

Reviewer: ```suggestion
To start using the new feature, add the `<defaultProperties>` configuration element to the `<jpa>` configuration in your `server.xml` file. Specify the persistence properties that you want to apply to all container-managed persistence units, as shown in the following examples:
```

  • specification in persistence.xml
```

> These `defaultProperties` persistence property values override any  properties  with the same name that are specified in a `persistence.xml` file. However, property values specified through PersistenceContext annotation, or the `persistence-context-ref` deployment descriptor element, or Query Hints property will override these `defaultProperties`.

Reviewer: ```suggestion
These `defaultProperties` persistence property values override any properties with the same name that are specified in a `persistence.xml` file. However, property values specified through PersistenceContext annotation, or the `persistence-context-ref` deployment descriptor element, or Query Hints property will override these `defaultProperties`.
```
